lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/protobuf/ProtocolBufferClassHierarchy.java

```diff
@@ -384,13 +385,39 @@ public ClassHierarchy merge(ClassHierarchy ch) {
     if (this == ch) {
       return this;
     }
-    throw new UnsupportedOperationException(
-        "Cannot merge ExternalClassHierarchies yet!");
+    if (!(ch instanceof ProtocolBufferClassHierarchy)) {
+      throw new UnsupportedOperationException(
```

Comment:
Throw a more specific error, like `"Cannot merge with class hierarchies of type: " + ch.class.getName()` or such.